crates/core/tedge_actors/src/message_boxes.rs

```diff
@@ -209,6 +212,68 @@ impl<Input: Message, Output: Message> SimpleMessageBox<Input, Output> {
     }
 }
 
+#[async_trait]
+impl<Input: Send, Output> Foo<Input> for SimpleMessageBox<Input, Output> {
+    async fn a(&mut self) -> Result<Option<Input>, RuntimeRequest> {
+        self.receiver.a().await
+    }
+
+    async fn b(&mut self) -> Option<WrappedInput<Input>> {
+        self.receiver.b().await
+    }
+
+    async fn c(&mut self) -> Option<Input> {
+        self.receiver.c().await
+    }
+}
+
+pub struct CombinedReceiver<Input> {
+    input_receiver: mpsc::Receiver<Input>,
+    signal_receiver: mpsc::Receiver<RuntimeRequest>,
+}
+
+impl<Input> CombinedReceiver<Input> {
+    pub fn new(
+        input_receiver: mpsc::Receiver<Input>,
+        signal_receiver: mpsc::Receiver<RuntimeRequest>,
+    ) -> Self {
+        Self {
+            input_receiver,
+            signal_receiver,
+        }
+    }
+}
+
+// TODO: log messages
+#[async_trait]
+impl<Input: Send> Foo<Input> for CombinedReceiver<Input> {
+    async fn a(&mut self) -> Result<Option<Input>, RuntimeRequest> {
+        match self.b().await {
+            Some(WrappedInput::Message(message)) => Ok(Some(message)),
+            Some(WrappedInput::RuntimeRequest(runtime_request)) => Err(runtime_request),
+            None => Ok(None),
+        }
+    }
+
+    async fn b(&mut self) -> Option<WrappedInput<Input>> {
```

Comment:
This is semanticly different to the current recv

Reply:
> This behaves differently to the current recv as when either channel is closed it returns None

I'm okay with that. One just has to keep in mind is that no runtime should be mandatory during tests. Here, one needs to provide a signal receiver that has at least one sender alive.